### PR TITLE
Created Account Activation Pin Form + Minimum Payment

### PR DESCRIPTION
**Pull request by williammfu**
## Type of changes

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

There is no pin form input and minimum payment modal on face payment solution

## What is the new behavior?

A PinForm component is provided for user 6-digit pin input on the third step and a popup modal is provided to determine the minimum payment. User can drag the slider or fill the field to determine their minimum payment.

## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/weber/blob/main/CHANGELOG.md)

## Security Pentester API Checklist
- [ ] IDOR
- [ ] Malicious Script Injection
- [ ] SQL Injection

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
image
image
image

### CODE REVIEW

File: frontend/app/api/paymentAPI.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import axios, { AxiosError } from 'axios'
 import { SESSION_ID_ERROR } from '../constants/message'
-import { PhoneNumberResponse } from '../types/responses'
+import { ActivationResponse, PhoneNumberResponse } from '../types/responses'
 
-type RegistrationResponse = {
+type PaymentResponse = {
```
Comment (rizkybaihaqy) [inline frontend/app/api/paymentAPI.ts:5]: pindahin ke type aja fu

File: frontend/app/components/modules/ActivationForm/ActivationForm.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import styles from './ActivationForm.module.scss'
+import { Color } from '@/types/elements'
+import { PaymentSetup } from '../PaymentSetup/PaymentSetup'
+import { parseCookies } from 'nookies'
+import { activateAccount } from '@/api/paymentAPI'
+
+type Props = {
+  nextStep: Function
+}
+const PIN_DIGIT_LENGTH = 6
```
Comment (rizkybaihaqy) [inline frontend/app/components/modules/ActivationForm/ActivationForm.tsx:15]: di pindah ke file konstant ae fu

**Comment by gregoriusjimmy** [comment]
image
kalau gw apus (pakai backspace) cursornya pindah kotak tapi ga ngehapus titiknya

File: frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  const handleInputChange = (e: BaseSyntheticEvent) => {
+    const typed = e.target.value
+    if (typed === '') {
+      setTypedInput(e.target.value)
+    } else if (/^[0-9\b]+$/.test(typed)) {
+      setTypedInput(e.target.value)
+      if (+typed >= MIN_PAYMENT && +typed <= MAX_PAYMENT) {
+        setValue(+typed)
+      }
```
Comment (gregoriusjimmy) [inline frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]: ini casenya kalau gw pilih next kyak gini (gambar) yang ke set tetep 300rb ya ?
image

Comment (krismp) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]: @gregoriusjimmy kak jim, nanya dong, input nominal nya disini bakal di format pake currency gak? mksdnya jadi nambah sendiri . di tiap 3 digit 0

Comment (krismp) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]: eh salah kak @williammfu ya wkwkwk

Comment (gregoriusjimmy) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]: wkwkw iya rencana juga butuh fungsi convert itu pas dibagian cart

File: frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import { BaseSyntheticEvent, useEffect, useState } from 'react'
+import styles from './PaymentSetup.module.scss'
+
+const MIN_PAYMENT = 50000
```
Comment (krismp) [inline frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:4]: ini masukin ke `env` aja kali ya, takut ada perubahan di masa depan, gak perlu rebuild docker imagenya, cukup ganti value dari .env nya aj, gimana @williammfu mnurut lu?

File: frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx (hunk 3 above)
Comment (krismp) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]: pake semacem component kaya gini ya https://www.npmjs.com/package/react-currency-input-field ? atau mau bikin sndiri trus di open source? ggwp

Comment (gregoriusjimmy) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]: kalau gitu cuman sebatas input field kah?

Comment (gregoriusjimmy) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]:
```js
export const formatMoney = money => {
  if (money === null) {
    return "0";
  }
  let number_string = money.toString(),
    sisa = number_string.length % 3,
    rupiah = number_string.substr(0, sisa),
    ribuan = number_string.substr(sisa).match(/\d{3}/g);

  if (ribuan) {
    const separator = sisa ? "." : "";
    rupiah += separator + ribuan.join(".");
  }
  if (money < 0) {
    return `-${rupiah}`;
  }
  return rupiah;
};

export const formatMoneyOnChange = money => {
  let number_string = money.replace(/[^,\d]/g, "").toString(),
    split = number_string.split(","),
    sisa = split[0].length % 3,
    rupiah = split[0].substr(0, sisa),
    ribuan = split[0].substr(sisa).match(/\d{3}/gi);

  // tambahkan titik jika yang di input sudah menjadi angka ribuan
  if (ribuan) {
    const separator = sisa ? "." : "";
    rupiah += separator + ribuan.join(".");
  }

  rupiah = split[1] !== undefined ? rupiah + "," + split[1] : rupiah;
  return rupiah;
};
```
ini dulu dari tugas kuliah, tolong cek lagi deh

Comment (krismp) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]: lgtm, mungkin tgl translate variable sm comment nya pake bahasa ~jaksel~ inggris aja kali ya @gregoriusjimmy , sama masukin ke helper atau lib? wdyt ? cc @williammfu @rizkybaihaqy

Comment (gregoriusjimmy) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]: yes masukin ke folder utils

Comment (krismp) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]: she cut klo gitu, bedain issue dan PR aja kali ya, biar gak nge blocking PR ini

Comment (krismp) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]: thank, ntap @gregoriusjimmy , GOKIL

Comment (gregoriusjimmy) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:30]: GOKIL @asyafalni

File: frontend/app/api/paymentAPI.ts (hunk 1 above)
Comment (williammfu) [reply frontend/app/api/paymentAPI.ts]: resolved on commit 5f010c3

File: frontend/app/components/modules/ActivationForm/ActivationForm.tsx (hunk 2 above)
Comment (williammfu) [reply frontend/app/components/modules/ActivationForm/ActivationForm.tsx]: resolved on commit 5f010c3

File: frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx (hunk 3 above)
Comment (williammfu) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx:26]: > ini casenya kalau gw pilih next kyak gini (gambar) yang ke set tetep 300rb ya ?
iya jim wkwk, keanya coba gue edit biar tulisan gede sm field sinkron deh

Comment (williammfu) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx]: field input udah sinkron sama display on commit 8b495a4

File: frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx (hunk 4 above)
Comment (gregoriusjimmy) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx]: gimana fu mnurut lu @williammfu

Comment (williammfu) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx]: suree boljugg

Comment (williammfu) [reply frontend/app/components/modules/PaymentSetup/PaymentSetup.tsx]: resolved on commit f8b9752

**Review by rizkybaihaqy — APPROVED** [review]

**Review by gregoriusjimmy — APPROVED** [review]
Tested, magnificent!

**Review by krismp — APPROVED** [review]
looks gokil to me

**Pull request merged**